### ISSUE TITLE
How to specify the parameter type in a form field?

### ISSUE THREAD

**Issue by wallace-pinho** [nelmio/NelmioApiDocBundle#1832]
I'm documenting an API and I'm using a FormType as the request body. Almost everything is working as expected, but when I have a parameter that is supposed to be an annotated entity class, it always use the type specified in the "add" method. 
e.g.

```php
public function buildForm(FormBuilderInterface $builder, array $options){
    $builder->add('my_property', \GenericType::class, array(
        'required' => false,
        'error_bubbling' => true,
        'data_class' => 'PathToAnnotatedEntityClass',
        'property' => 'my_property',
        'documentation' => [
            'ref' => new \Nelmio\ApiDocBundle\Annotation\Model(["type" => "PathToAnnotatedEntityClass", "groups" => ["all"]]),
        ],
    ));
}
```

The bundle then generates the docs using "GenericType" for "my_property". Is there a way to specify an annotated entity class to be used instead? I tried using the documentation option but it doesn't seem to work.
Please, tell me if I was not clear enough or if I need to be more specific about my issue.

**Comment by GuilhemN** [comment]
Are you using NelmioApiDocBundle v3 or v4 ?

For both v3 and v4 you can use `'documentation' => ['type' => 'object', 'schema' => ['ref' => '#pathtoyourmodel']]`.
In fact only `type` was detected so far, but that's not very good when you want to use a model indeed.

https://github.com/nelmio/NelmioApiDocBundle/pull/1834 should make the example you provided work for NelmioApiDocBundle v4. Could you check if that indeed solves your issue in case you're using v4?

**Comment by wallace-pinho** [comment]
My version of NelmioApiDocBundle is locked at 4.0.1 because of some dependency I can't remember right now. 
Anyway, thank you for your reply. I'll test it as soon as I have time and I'll let you know if it works.

**Comment by wallace-pinho** [comment]
@GuilhemN #1834 is not working for me. It throw the following error (I've cropped the stacktrace): 
```
TypeError:
substr() expects parameter 1 to be string, object given

  at vendor/zircote/swagger-php/src/Annotations/AbstractAnnotation.php:401
  at substr(object(Model), 0, 2)
     (vendor/zircote/swagger-php/src/Annotations/AbstractAnnotation.php:401)
  at OpenApi\Annotations\AbstractAnnotation->validate(array(object(OpenApi), object(Components), object(Schema)), array(object(OpenApi), object(Components), object(Schema)), '#/components/schemas/2/properties/7')
     (vendor/zircote/swagger-php/src/Annotations/Schema.php:381)
  at OpenApi\Annotations\Schema->validate(array(object(OpenApi), object(Components), object(Schema)), array(object(OpenApi), object(Components), object(Schema)), '#/components/schemas/2/properties/7')
     (vendor/zircote/swagger-php/src/Annotations/AbstractAnnotation.php:482)
```

Looking at the code I came up with a solution that solves my problem. I opened #1836. Please, take a look and let me know what you think about it.

**Comment by GuilhemN** [comment]
Sorry for not answering your PR @wallace-pinho.

I was planning on fixing my PR which fixes the issue in a more generic way but then did not find the time to do so.
I pushed new commits and a test case to ensure your use case is covered.

Closing as I just merged #1834 to solve this.
